File: DSA/Big O/JS/06-Big-O-Problem.js
function funChallenge(input){
    let a = 15; // O(1) constant
    a = 40+5; // O(1) constant

    for (let i=0; i<input.length; i++){ // O(n) linear
        anotherFunction(); // O(n) linear
        let newVariable = true; // O(n) linear
        a++; // O(n) linear
    }
    return a; // O(1) constant
}

// O(1)+O(1)+O(n)+O(n)+O(n)+O(n)+O(1)
// 1+1+n+n+n+n+1 = 3+4n = Big O(3+4n) = O(n) linear time

funChallenge() // O(n)

File: DSA/Big O/Python/06-Big-O-Problem.py
def funChallenge(input):
    x = 5 # O(1)
    y = x+10 # O(1)
    z = x*2 # O(1)

    for i in range(10): # O(n)
        str(y) # O(n)
        str(z) # O(n)
        p = "This is new string" # O(n)
        x+=1 # O(n)

    return "hello world" # O(1)

# O(1)+O(1)+O(1)+O(n)+O(n)+O(n)+O(n)+O(n)+O(1)
# 1+1+1+n+n+n+n+n+1 = 4+5n = Big O(4+5n) = O(n)

funChallenge(10) # O(n)
